src/app/search/search.component.ts

```diff
@@ -8,9 +8,11 @@ import { Component } from '@angular/core';
 export class SearchComponent {
 
   public searchText: string;
+  @Output() filterCourses: EventEmitter<string> = new EventEmitter<string>();
 
   onSearch(): void {
-    console.log(`The search text is ${this.searchText ? this.searchText : 'empty'}`);
+    this.filterCourses.emit(this.searchText || '');
+    !this.searchText && console.log(`The search text is empty`);
   }
```

Comment:
In this case it can be simplified (early return approach):
```typescript
if (!this.searchText) {
  return;
}

this.filterCourses.emit(this.searchText);
```